lldb/source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -2285,34 +2306,19 @@ GetFollowAsyncContextUnwindPlan(RegisterContext *regctx, ArchSpec &arch,
   const int32_t ptr_size = 8;
   row->SetOffset(0);
 
-  uint32_t async_context_regnum;
-  uint32_t fp_regnum;
-  uint32_t pc_regnum;
-  uint32_t dummy_regnum;
-
-  if (arch.GetMachine() == llvm::Triple::x86_64) {
-    async_context_regnum = dwarf_r14_x86_64;
-    fp_regnum = dwarf_rbp_x86_64;
-    pc_regnum = dwarf_rip_x86_64;
-    dummy_regnum = dwarf_r15_x86_64;
-  } else if (arch.GetMachine() == llvm::Triple::aarch64) {
-    async_context_regnum = arm64_dwarf::x22;
-    fp_regnum = arm64_dwarf::fp;
-    pc_regnum = arm64_dwarf::pc;
-    dummy_regnum = arm64_dwarf::x23;
-  } else {
+  auto registers = GetAsyncUnwindRegisters(arch.GetMachine());
```

Comment:
This is 100% a personal choice, but I think it makes it harder to read the code years from now with this kind of `auto` use.  Is this getting a register value?  Is it getting a register context?  What can I do with the returned thing, is it a pointer that can return nullptr?  Sometimes auto is a billion times better like for (auto it = vec.begin(), but sometimes I think it adds a burden to future maintainers who are reading it. If this were a common idiom that we saw, like auto target_sp(GetTarget()), instead of TargetSP, everyone would know what this was doing.

I'm not saying you should change this, just throwing out my personal opinion and I think I am not in the majority within the group, so feel free to leave it as-is.

Reply:
changed it, thanks